server/bin/pbench-verify-backup-tarballs.py

```diff
@@ -290,38 +290,41 @@ def main():
 
     logger = get_pbench_logger(_NAME_, config)
 
-    archive = Path(config.ARCHIVE).resolve()
-    if not archive:
+    archive = Path(config.ARCHIVE)
+    try:
+        archivepath = archive.resolve(strict=True)
+    except FileNotFoundError:
         logger.error(
             "The ARCHIVE directory {}, does not resolve to a real location", archive
         )
         return 1
-
-    if not archive.is_dir():
-        logger.error(
-            "The setting for ARCHIVE in the config file is {}, but that is"
-            " not a directory",
-            archive,
-        )
-        return 1
+    else:
+        if not archivepath.is_dir():
+            logger.error(
+                "The setting for ARCHIVE in the config file is {}, but that is"
+                " not a directory",
+                archive,
+            )
+            return 1
 
     # add a BACKUP field to the config object
     config.BACKUP = backup = config.conf.get("pbench-server", "pbench-backup-dir")
```

Comment:
The error logged below is for this line (line 311), so we'll need a try/except here and log that error in the exception.

Reply:
+1

If we get an empty string from the `config.conf.get()` invocation, then the `Path()` constructor will just use the current working directory to make the `Path` object.  So we need to check that we actually got a value from the config file first.

We might mean we have the same bug in the `PbenchServerConfig` class constructor.